src/lib/serviceRoutines/postQueryContext.cpp

```diff
@@ -86,6 +86,35 @@ static char* jsonPayloadClean(const char* payload)
 
 
 
+/* ***************************************************************************
+*
+* getProviderCount -
+*
+* getProviderCount is created for extrating the 'Fiware-Total-Count' header value so that we can add the same into total count,
+* And we can get the actualTotalCount value that includes total Entities of CP+CB.
+*
```

Comment:
I'd suggest the potential improvement in the logic we have discussed [previously](https://github.com/telefonicaid/fiware-orion/pull/4275#discussion_r1195415590)

```suggestion
*
* FIXME: this functions relies in text-based processing of the HTTP response stream (which includes headers).
* It would be smarted to parse all the response headers in the response (for instance in a std::map) in the part
* of the code dealing with response processing, so in this part we can just access to 'fiware-total-count'  easility
* using something like reponseHeaders["fiware-total-count"]
```